File: server/src/sentry.ts
import * as Sentry from "@sentry/node"
import { nodeProfilingIntegration } from "@sentry/profiling-node"
import type { ServerConfig } from "./types.js"

/**
 * Initializes Sentry for the Node.js Cloud Run server.
 */
export function initSentry(config: ServerConfig): void {
    if (!config.sentryDsn) {
        return
    }

    Sentry.init({
        dsn: config.sentryDsn,
        environment: config.sentryEnvironment,
        release: config.sentryRelease,
        integrations: [nodeProfilingIntegration()],
        tracesSampleRate: 1.0,
        profilesSampleRate: 1.0,
        debug: true
    })
}

/**
 * Sends a debug event to Sentry with structured context.
 */
export function captureDebugMessage(message: string, extra: Record<string, unknown>): void {
    Sentry.captureMessage(message, {
        level: "debug",
        extra
    })
}

/**
 * Captures unexpected runtime exceptions in Sentry.
 */
export function captureException(error: unknown): void {
    Sentry.captureException(error)
}

File: worker/chatrag_worker/sentry.py
import sentry_sdk
from .config import WorkerConfig


def init_sentry(config: WorkerConfig) -> None:
    """Initialize Sentry worker SDK when DSN is configured."""
    if not config["sentry_dsn"]:
        return

    sentry_sdk.init(
        dsn=config["sentry_dsn"],
        environment=config["sentry_environment"],
        release=config["sentry_release"],
        traces_sample_rate=1.0,
        debug=True,
    )


def capture_debug(message: str, **extra: object) -> None:
    """Send worker debug message to Sentry."""
    sentry_sdk.set_context("debug", extra)
    sentry_sdk.capture_message(message, level="debug")
